externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkProgressFetcher.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.kyuubi
+
+import org.apache.hive.service.rpc.thrift.TProgressUpdateResp
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.status.api.v1.StageStatus
+
+import org.apache.kyuubi.engine.spark.operation.progress.{SparkProgressMonitor, SparkStage, SparkStageProgress}
+
+class SparkProgressFetcher(spark: SparkSession, jobGroup: String) {
+
+  val statusStore = spark.sparkContext.statusStore
+
+  def getJobProgressUpdate(startTime: Long): TProgressUpdateResp = {
+    val progressMap = getProgressMap
+    val progressMonitor = new SparkProgressMonitor(progressMap)
+    new TProgressUpdateResp(
+      progressMonitor.headers,
+      progressMonitor.rows,
+      progressMonitor.progressedPercentage,
+      progressMonitor.executionStatus,
+      progressMonitor.footerSummary,
+      startTime)
+  }
+
+  private def getProgressMap(): Map[SparkStage, SparkStageProgress] = {
```

Comment:
move this to SparkProgressMonitor as a `private val`?

and we sort it once?
https://github.com/apache/incubator-kyuubi/pull/2529/files#diff-441ca5a194460cd9dfe41b5e3db54989c91d9aac23418329b1b3216ba7d8c7eeR32
https://github.com/apache/incubator-kyuubi/pull/2529/files#diff-441ca5a194460cd9dfe41b5e3db54989c91d9aac23418329b1b3216ba7d8c7eeR76

Reply:
How about removing SparkProgressFetcher and merging it into SparkProgressMonitor?

Reply:
> How about removing SparkProgressFetcher and merging it into SparkProgressMonitor?

Or just make getProgressMap return SortedMap

Reply:
> How about removing SparkProgressFetcher and merging it into SparkProgressMonitor?

+1